mne/io/edf/edf.py

```diff
@@ -511,16 +471,547 @@ def _get_edf_info(fname, stim_channel, annot, annotmap, eog, misc, exclude,
+        lowpass = np.ravel([re.findall('LP:\s+(\w+)', filt)
+                            for filt in prefiltering])
+
+        # number of samples per record
+        n_samps = np.array([int(fid.read(8).decode()) for ch
+                            in channels])
+
+        # Populate edf_info
+        edf_info['ch_names'] = ch_names
+        edf_info['data_offset'] = header_nbytes
+        edf_info['date'] = date
+        edf_info['digital_max'] = digital_max
+        edf_info['digital_min'] = digital_min
+        edf_info['exclude'] = exclude
+        edf_info['highpass'] = highpass
+        edf_info['include'] = include
+        edf_info['lowpass'] = lowpass
+        edf_info['n_records'] = n_records
+        edf_info['n_samps'] = n_samps
+        edf_info['nchan'] = nchan
+        edf_info['physical_max'] = physical_max
+        edf_info['physical_min'] = physical_min
+        edf_info['record_length'] = record_length
+        edf_info['subtype'] = subtype
+
+        fid.read(32 * nchan).decode()  # reserved
+        assert fid.tell() == header_nbytes
+
+        fid.seek(0, 2)
+        n_bytes = fid.tell()
+        n_data_bytes = n_bytes - header_nbytes
+        total_samps = (n_data_bytes // 3 if subtype == '24BIT'
+                       else n_data_bytes // 2)
+        read_records = total_samps // np.sum(n_samps)
+        if n_records != read_records:
+            warn('Number of records from the header does not match the file '
+                 'size (perhaps the recording was not stopped before exiting).'
+                 ' Inferring from the file size.')
+            edf_info['n_records'] = n_records = read_records
+
+        if subtype in ('24BIT', 'bdf'):
+            edf_info['data_size'] = 3  # 24-bit (3 byte) integers
+        else:
+            edf_info['data_size'] = 2  # 16-bit (2 byte) integers
+
+    return edf_info
+
+
+def _read_gdf_header(fname, stim_channel, exclude):
+    """Read GDF 1.x and GDF 2.x header info."""
+    edf_info = dict()
+    # edf_info['annot'] = annot
+    # edf_info['annotmap'] = annotmap
+    edf_info['events'] = []
+
+    with open(fname, 'rb') as fid:
+        assert(fid.tell() == 0)
+
+        version = fid.read(8).decode()
+
+        __gdftyp_byte = (1, 1, 1, 2, 2, 4, 4, 8, 8, 4, 8, 0, 0, 0, 0, 0, 4, 8,
+                         16)
+        __gdftyp_np = [None]
+        __gdftyp_np.append(np.int8)
+        __gdftyp_np.append(np.uint8)
+        __gdftyp_np.append(np.int16)
+        __gdftyp_np.append(np.uint16)
+        __gdftyp_np.append(np.int32)
+        __gdftyp_np.append(np.uint32)
+        __gdftyp_np.append(np.int64)
+        __gdftyp_np.append(np.uint64)
+        __gdftyp_np.append(None)
+        __gdftyp_np.append(None)
+        __gdftyp_np.append(None)
+        __gdftyp_np.append(None)
+        __gdftyp_np.append(None)
+        __gdftyp_np.append(None)
+        __gdftyp_np.append(None)
+        __gdftyp_np.append(np.float32)
+        __gdftyp_np.append(np.float64)
```

Comment:
... and you can do this, too:
```
__gdftyp_byte = [np.dtype(x).itemsize if x is not None else 0 for x in __gdftyp_np]
```
And to sanity check you can add something like:
```
assert sum(__gdftyp_byte) == ...
```
Just to sanity check (which is nice).

Also we don't prefix variables with `__`. You don't need them, `gdftype_byte` is fine already.